sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,1013 @@
+        impostor pairs have to be identified explicitly.
+
+    max_iter : int, optional (default=50)
+        Maximum number of iterations in the optimization.
+
+    tol : float, optional (default=1e-5)
+        Convergence tolerance for the optimization.
+
+    max_corrections : int, optional (default=100)
+        The maximum number of variable metric corrections
+        used to define the limited memory matrix. (The limited memory BFGS
+        method does not store the full hessian but uses this many terms in an
+        approximation to it.)
+
+    callback : callable, optional (default=None)
+        If not None, this function is called after every iteration of the
+        optimizer taking as arguments the current solution and the number of
+        iterations.
+
+    verbose : int, optional (default=0)
+        If 0, no progress messages will be printed.
+        If 1, progress messages will be printed to stdout.
+        If >1, progress messages will be printed and the ``iprint``
+        parameter of :meth:`fmin_l_bfgs_b` of `scipy.optimize` will be set to
+        verbose - 2.
+
+    random_state : int or numpy.RandomState or None, optional (default=None)
+        A pseudo random number generator object or a seed for it if int.
+
+    n_jobs : int, optional (default=1)
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Doesn't affect :meth:`fit` method.
+
+    Attributes
+    ----------
+    transformation_ : array, shape (n_features_out, n_features).
+        The linear transformation learned during fitting.
+
+    n_neighbors_ : int
+        The provided n_neighbors is decreased if it is greater than or equal
+        to  min(number of elements in each class).
+
+    classes_inverse_non_singleton_ : array, shape (n_classes_non_singleton,)
+        The appearing classes that have more than one sample, transformed to
+        be integers within the range(0, n_classes).
+
+    n_funcalls_ : int
+        Counts the number of times the loss and the gradient are computed.
+
+    n_iter_ : int
+        Counts the number of iterations performed by the optimizer.
+
+    details_ : dict
+        A dictionary of information returned by the L-BFGS optimizer,
+        augmented with the final loss value and the measured training time.
+
+        * details_['warnflag'] is
+
+          - 0 if converged,
+          - 1 if too many function evaluations or too many iterations,
+          - 2 if stopped for another reason, given in d['task']
+
+        * details_['grad'] is the gradient at the minimum (should be 0 ish)
+        * details_['funcalls'] is the number of function calls made.
+        * details_['nit'] is the number of iterations.
+        * details_['loss'] is the the objective value at the solution found.
+        * details_['time'] is the time it took to train the model in seconds.
```

Comment:
we usually do not store all these details eg in logisticregression that uses L-BFGS too. I doubt the vast majority of our users don't need all these details.

Reply:
let's not blindly copy. We should focus on usecases and user expectations. I mean the expectations of the 90% of our users so I think it's an overkill.

Reply:
I understand. How about a `store_details` parameter set to False by default?